### PR TITLE
common: move big integer math to common/math

### PR DESCRIPTION
**Pull request by fjl**
This PR:

* deletes dead code in packages common and common/math
* moves all big integer math, decoding, parsing to common/math
* tightens integer parsing rules for genesis block JSON and cli flags

I started working on this as prep-work for config file support but couldn't help myself after finding so much sloppy big.Int code. The 'common big integers often used' are left in package common for now.

A bit more about the tighter parsing: Previously a command like `geth --gasprice=ggg --gpomin=0123` specified a gas price of zero and a gas price oracle floor of 83 (leading zero switched parsing to octal). The new code errors for invalid numbers and parses numbers with leading zero as decimal.

### CODE REVIEW

**Comment by fjl** [comment]
Note: I want to squash this myself.

**Comment by fjl** [comment]
@holiman please check again, I added one more check

File: common/README.md [hunk 1]
```diff
@@ -1,140 +0,0 @@
-# common
```
Comment (obscuren) [inline common/README.md]: Hah, wow this is old!

What an ancient relic you've found! Even the travis file contains `go 1.2`!

**Comment by fjl** [comment]
This is now rebased. We should just squash it to one commit otherwise individual commits won't build.

**Pull request merged**